### PR TITLE
improve SslStream tests on server 2022

### PR DESCRIPTION
**Pull request by wfurt**
It seems like server 2022 also has TLS 1.3 enabled by default and it has similar issue like Windows 11. 

This updates some PlatformDetection references to work as intended. 

cc: @safern

### CODE REVIEW

File: src/libraries/System.Net.Security/tests/FunctionalTests/SslStreamNetworkStreamTest.cs [hunk 1]
```diff
@@ -472,17 +472,12 @@ public async Task SslStream_NegotiateClientCertificateAsync_PendingDecryptedData
         }
 
         [ConditionalTheory(typeof(PlatformDetection), nameof(PlatformDetection.SupportsTls13))]
+        [SkipOnPlatform(TestPlatforms.Windows, "unstable on Windows")]
```
Comment (antonfirsov) [inline src/libraries/System.Net.Security/tests/FunctionalTests/SslStreamNetworkStreamTest.cs:475]:
```suggestion
        [ActiveIssue("https://github.com/dotnet/runtime/issues/58927", TestPlatforms.Windows)]
```

**Review by stephentoub — APPROVED** [review]

**Review by safern — APPROVED** [review]

**Comment by wfurt** [comment]
Quic failures is https://github.com/dotnet/runtime/issues/64944. rest looks like infrastructure problem.

**Pull request merged**